wgengine/monitor/monitor.go

```diff
@@ -18,6 +18,13 @@ import (
 	"tailscale.com/types/logger"
 )
 
+// pollWallTimeInterval is how often we check the time to check
+// for big jumps in wall (non-monotonic) time as a backup mechanism
+// to get notified of a sleeping device waking back up.
+// Usually there are also minor network change events on wake that let
+// us check the wall time sooner than this.
+const pollWallTimeInterval = 15 * time.Second
```

Comment:
Does time.AfterFunc operate on wall time or monotonic time?

That is, on waking from a sleep of 1m, will this time fire immediately or potentially only after 15s?

I assume the latter, which is unfortunate, as there's then tension between waking up all the time to do (mostly) useless work vs being fast to respond on wake-from-sleep.

I wonder whether we should exclude mobile platforms from this?

I am still OK with the use of a big hammer, but I am less keen on it than I was initially.

Another option is to use something more targeted per-platform. You can register for wake/sleep notifications on macOS from C: https://developer.apple.com/library/archive/qa/qa1340/_index.html and https://gist.github.com/nicolai86/fd720c9a660188fe29d7ac7a810f96f3.

Reply:
AfterFunc etc are monotonic, not wall. I picked 15 seconds as a sort of compromise between being annoying to CPU and human patience.

I plan to also hook in OS-specific wake events. I started on that already in in corp f5d42e6297c92d6490045746732df5eca10fe295 from the other side. I then add a method to this to just re-check the wall time and nothing else.

I'll submit this first, though.

Reply:
Turned off ios/android for now, though. They don't add much compared to the other platforms, as ios/android never sleep in the same way.